src/fosslight_util/write_opossum.py

```diff
@@ -285,11 +287,14 @@ def make_resources_and_attributions(sheet_items, scanner, resources, fc_list):
             resourcesToAttributions[os.path.join(os.sep, path)] = uuid_list
 
             for ext in externalAttribution_list:
-                dict = ext.get_externalAttribution_dict()
-                externalAttributions[str(ext.uuid)] = dict
+                externalAttributions[str(ext.uuid)] = ext
     except Exception as e:
         logger.error("Failed to make_resources_and_attributions: " + str(e))
         logger.error(traceback.format_exc())
         success = False
 
-    return success, resources, externalAttributions, resourcesToAttributions, fc_list, ab_list
+    externalAttributionsConverted = {
+        uuid: ext.get_externalAttribution_dict() for uuid, ext in externalAttributions.items()
```

Comment:
`dataclasses` would be very useful for a lot of things which happen in this code.
You can make them frozen and thus hashable, can exclude fields for the comparison/object equality, convert them to a dict with `asdict`.

Reply:
I searched the `dataclasses` and it was added in python3.7.
Now FOSSLight scanners support python3.6+.
We will consider it when we update the supported python version later.